src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.cs

```diff
@@ -66,13 +67,13 @@ public virtual JsonTypeInfo GetTypeInfo(Type type, JsonSerializerOptions options
             Justification = "The ctor is marked RequiresDynamicCode.")]
         private JsonTypeInfo CreateJsonTypeInfo(Type type, JsonSerializerOptions options)
         {
-            MethodInfo methodInfo = typeof(JsonSerializerOptions).GetMethod(nameof(JsonSerializerOptions.CreateReflectionJsonTypeInfo), BindingFlags.NonPublic | BindingFlags.Instance)!;
+            MethodInfo methodInfo = typeof(DefaultJsonTypeInfoResolver).GetMethod(nameof(CreateReflectionJsonTypeInfo), BindingFlags.NonPublic | BindingFlags.Static)!;
 #if NETCOREAPP
-            return (JsonTypeInfo)methodInfo.MakeGenericMethod(type).Invoke(options, BindingFlags.NonPublic | BindingFlags.DoNotWrapExceptions, null, null, null)!;
+            return (JsonTypeInfo)methodInfo.MakeGenericMethod(type).Invoke(null, BindingFlags.NonPublic | BindingFlags.DoNotWrapExceptions, null, new[] { options }, null)!;
 #else
             try
             {
-                return (JsonTypeInfo)methodInfo.MakeGenericMethod(type).Invoke(options, null)!;
+                return (JsonTypeInfo)methodInfo.MakeGenericMethod(type).Invoke(null, new[] { options })!;
```

Comment:
previously this didn't allocate on every usage, now it does. I think it's better to keep CreateReflectionJsonTypeInfo in JsonSerializerOptions, even if it's a weird place to keep it

Reply:
You mean the argument array? Given this is a reflection call, the added allocation hardly registers w.r.t. cost. In any case, creating `JsonTypeInfo` instances is not in the hot path.